snow/validators/manager.go

```diff
@@ -22,25 +23,83 @@ var (
 	ErrMissingValidators = errors.New("missing validators")
 )
 
+type SetCallbackListener interface {
+	OnValidatorAdded(validatorID ids.NodeID, pk *bls.PublicKey, txID ids.ID, weight uint64)
+	OnValidatorRemoved(validatorID ids.NodeID, weight uint64)
+	OnValidatorWeightChanged(validatorID ids.NodeID, oldWeight, newWeight uint64)
+}
+
 // Manager holds the validator set of each subnet
 type Manager interface {
 	fmt.Stringer
 
-	// Add a subnet's validator set to the manager.
-	//
-	// If the subnet had previously registered a validator set, false will be
-	// returned and the manager will not be modified.
-	Add(subnetID ids.ID, set Set) bool
+	// Add a new staker to the subnet.
+	// Returns an error if:
+	// - [weight] is 0
+	// - [nodeID] is already in the validator set
+	// If an error is returned, the set will be unmodified.
+	AddStaker(subnetID ids.ID, nodeID ids.NodeID, pk *bls.PublicKey, txID ids.ID, weight uint64) error
```

Comment:
q: we don't regiqster subnetIDs independently anymore. Should we return a boolean to signal where a new subnet has been added or whether validator belongs to a subnet already seen?

Reply:
I'm not sure we'll ever use the boolean... So I feel like we can go without and add it if needed later